instrumentation/quarkus-resteasy-reactive/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/quarkus/resteasy/reactive/OtelRequestContext.java

```diff
@@ -0,0 +1,43 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.javaagent.instrumentation.quarkus.resteasy.reactive;
+
+import io.opentelemetry.instrumentation.api.instrumenter.http.HttpRouteSource;
+import org.jboss.resteasy.reactive.server.core.ResteasyReactiveRequestContext;
+
+public final class OtelRequestContext {
+  private static final ThreadLocal<OtelRequestContext> contextThreadLocal = new ThreadLocal<>();
```

Comment:
Would it make sense to use OTel `Context` instead of a thread local?

Reply:
I think the `ThreadLocal` uses slightly less code. I don't have a strong preference, if you want I can rewrite to otel context.